STM32F1/platform.txt

```diff
@@ -14,7 +14,7 @@ compiler.warning_flags.all=-Wall -Wextra -DDEBUG_LEVEL=DEBUG_ALL
 
 # compiler variables
 # ----------------------
-compiler.path={runtime.tools.arm-none-eabi-gcc.path}/bin/
+compiler.path={runtime.tools.xpack-arm-none-eabi-gcc-9.2.1-1.1.path}/bin/

```

Comment:
This path is generated and managed by the Arduino IDE in the same way as it is with the current toolchain.
When you install a package on the Arduino IDE, the JSON file describes all the boards and what toolchain each one uses.
Here's a partial of the STM32 Cores JSON file describing the toolchain and where the IDE should download it from:

![image](https://user-images.githubusercontent.com/50145413/82929590-d44f9300-9f7b-11ea-8aed-99123d35eb20.png)

Arduino IDE will then reference these tools by the "**name**" and will create a folder with that name inside the tools folder for that package. Then it will create another folder with the "**version**" as the name (so you can have different versions of the tools) and that's where the IDE will unpack the tools described in the JSON file.
Note that this JSON can be quite hard to digest as it typically describes all the versions of the package and each description is very, very long.

The _Wiki_ entry for **Installation** states (second bullet):
_Run the IDE, and on the Tools menu, select the Boards manager, and install the Arduino SAM boards (Cortex-M3) from the list of available boards. You must do this step, it installs the arm-none-eabi-g++ toolchain!_
And for this new toolchain, this step needs to change to **STM32 Cores** instead of **Arduino SAM boards**.

Reply:
You want to say that we need to install the STM32 core in order to use this PR?
I do not think that it is a viable solution, as this core should be independent form other cores.
This core is usually not installed using the Arduino package installer, but downloaded and extracted from ZIP.

Reply:
However you still have to use the Arduino package installer to install the SAMD boards, as stated in the Wiki. Otherwise you won't have the required toolchain. It's just a matter of installing a different package to use a different toolchain.

![image](https://user-images.githubusercontent.com/50145413/82957752-b437c800-9fab-11ea-9d60-7f0e26709313.png)

I'm investigating a way of having both toolchains available that I think will please you.

Reply:
Update: Just pushed a new commit that adds an option to change between xPack and the old arm-none-eabi-gcc .
I think this will make it easier to see the gains between using the new tools and the old ones and improve on that.

Reply:
I do not think that installing an additional **board package** (SAM boards) can be made equivalent to installing an additional **core**.

I think the users can download the new xpack separately and copy it to the old arm eabi directory, without installing a complete new core.

Reply:
It is exactly the same, @stevstrong. When someone installs the SAM boards package, that person is actually installing an additional full Core. Your suggestion is to (if someone wants to use more recent and better tools) potentially break their installation of the other Cores and Boards. Doesn't seem right.

Reply:
The SAM board package is ~250MB large, the STM32DUINO package is ~1 GB large, also 4 times larger. Alone form this point of view it makes a huge difference to have on or the other package...

Reply:
> Alone form this point of view it makes a huge difference to have on or the other package...

Yes CMSIS and arm gcc are huge.